esphome/components/at581x/at581x.cpp

```diff
@@ -0,0 +1,213 @@
+const uint8_t FREQ60_TABLE[] = {0x9d, 0x9d, 0x9d, 0x9d, 0x9d, 0x9d, 0x9d, 0x9d, 0x9e, 0x9e, 0x9e, 0x9e};
+
+/*!< Value for RF and analog modules switch (p10). */
+const uint8_t RF_OFF_TABLE[] = {0x46, 0xaa, 0x50};
+const uint8_t RF_ON_TABLE[] = {0x45, 0x55, 0xA0};
+const uint8_t RF_REG_ADDR[] = {0x5d, 0x62, 0x51};
+
+/*!< Registers of Lighting delay time. Unit: ms, min 2s (p8) */
+const uint8_t HIGH_LEVEL_DELAY_CONTROL_ADDR = 0x41; /*!< Time_flag_out_ctrl 0x01 */
+const uint8_t HIGH_LEVEL_DELAY_VALUE_ADDR = 0x42;   /*!< Time_flag_out_1 Bit<7:0> */
+
+const uint8_t RESET_ADDR = 0x00;
+
+/*!< Sensing distance address */
+const uint8_t SIGNAL_DETECTION_THRESHOLD_ADDR_LO = 0x10;
+const uint8_t SIGNAL_DETECTION_THRESHOLD_ADDR_HI = 0x11;
+
+/*!< Bit field value for power registers */
+const uint8_t POWER_THRESHOLD_ADDR_HI = 0x68;
+const uint8_t POWER_THRESHOLD_ADDR_LO = 0x67;
+const uint8_t PWR_WORK_TIME_EN = 8;     // Reg 0x67
+const uint8_t PWR_BURST_TIME_EN = 32;   // Reg 0x68
+const uint8_t PWR_THRESH_EN = 64;       // Reg 0x68
+const uint8_t PWR_THRESH_VAL_EN = 128;  // Reg 0x67
+
+/*!< Times */
+const uint8_t TRIGGER_BASE_TIME_ADDR = 0x3D;  // 4 bytes, so up to 0x40
+const uint8_t PROTECT_TIME_ADDR = 0x4E;       // 2 bytes, up to 0x4F
+const uint8_t TRIGGER_KEEP_TIME_ADDR = 0x42;  // 4 bytes, so up to 0x45
+const uint8_t TIME41_VALUE = 1;
+const uint8_t SELF_CHECK_TIME_ADDR = 0x38;  // 2 bytes, up to 0x39
+
+namespace esphome {
+namespace at581x {
+
+static const char *const TAG = "at581x";
+
+AT581XComponent::AT581XComponent() {}
+
+bool AT581XComponent::i2c_write_reg(uint8_t addr, uint8_t data) {
+  return write_register(addr, &data, 1) == esphome::i2c::NO_ERROR;
+}
+bool AT581XComponent::i2c_write_reg(uint8_t addr, uint32_t data) {
+  return i2c_write_reg(addr + 0, uint8_t(data & 0xFF)) && i2c_write_reg(addr + 1, uint8_t((data >> 8) & 0xFF)) &&
+         i2c_write_reg(addr + 2, uint8_t((data >> 16) & 0xFF)) && i2c_write_reg(addr + 3, uint8_t((data >> 24) & 0xFF));
+}
+bool AT581XComponent::i2c_write_reg(uint8_t addr, uint16_t data) {
+  return i2c_write_reg(addr, uint8_t(data & 0xFF)) && i2c_write_reg(addr + 1, uint8_t((data >> 8) & 0xFF));
+}
+
+bool AT581XComponent::i2c_read_reg(uint8_t addr, uint8_t &data) {
+  return read_register(addr, &data, 1) == esphome::i2c::NO_ERROR;
+}
+
+void AT581XComponent::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up AT581X...");  //, this->id_.c_str());
+  this->detection_pin_->setup();
+  if (!i2c_write_config()) {
+    ESP_LOGCONFIG(TAG, "Setting up AT581X failed...");  //, this->id_.c_str());
+  }
+}
+void AT581XComponent::loop() {
+  // The main operation is to detect the human presence
+  bool state = this->detection_pin_->digital_read();
+  if (this->motion_binary_sensor_ != nullptr) {
+    this->motion_binary_sensor_->publish_state(state);
+  }
+}
+void AT581XComponent::dump_config() {
+  if (this->motion_binary_sensor_ != nullptr) {
+    LOG_BINARY_SENSOR("", "AT581X", this->motion_binary_sensor_);
+  }
```

Comment:
Not required as the macro does this already
```suggestion
  LOG_BINARY_SENSOR("", "AT581X", this->motion_binary_sensor_);
```